app/scripts/services/rdp.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("RdpService", function(Logger) {
+    function findRDPPort (service, rdpPortNumber) {
+      var servicePorts = _.get(service, 'spec.ports');
+      return _.find(servicePorts, {targetPort: rdpPortNumber});
+    }
+
+    return {
+      buildRdp: function (address, port) {
+        return '' +
```

Comment:
This is not a terribly big deal, and looks like you already have a n LGTM, but I tend to think long complicated string concat is easier to maintain when an array:

```JavaScript 
return [
  'full address:s:' + address + ':' + port,   // fine to concat if this is a logical unit
  '\nusername:s:Administrator',
  '\nscreen mode id:i:2',
  '\nprompt for credentials:i:1',
  // ... etc ... 
].join('');
```

Reply:
More people you ask, more opinion on syntax you get. Fine for me, refactored as suggested.